airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValueDeepCoercingMapper.kt

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+import io.airbyte.cdk.load.util.serializeToString
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.OffsetTime
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+/**
+ * A mapper which coerces ALL values against the schema. This mapper MUST NOT be called after any
+ * mapper that returns non-native-JSON types (date, timestamp, etc.), or any mapper that causes the
+ * values to become misaligned with the schema (e.g. [AirbyteSchemaNoopMapper] +
+ * [SchemalessValuesToJsonString]).
+ *
+ * If this mapper is included in a [MapperPipeline], it SHOULD be preceded by a [MergeUnions]
+ * mapper. Not including this mapper may result in strange behavior when coercing union types.
```

Comment:
So if I'm following so far
Initial marshaling -> just turns the json nodes into their equivalent airbyte values, w/o respecting the schema
this mapper forces everything to align with the schema

so if i wanted to use this to aggressively filter in parquet

```
Marshal -> [Pipeline: MapTimestamps -> SchemalessToStrings -> MergeUnions -> Coerce -> ToDisjointRecord] -> ToAvro
```

Reply:
sounds right? not sure what you're putting into MapTimestamps, but if that's operating on raw StringValue, then yes

(and putting SchemalessToString before the Coerce mapper means that we need to also modify the schema to be StringType, unless we have some way to tell ToDisjointRecord that it was originally an ObjectWithEmptySchema/etc, though that's the thing we're trying to solve in slack )